curriculum/challenges/english/25-front-end-development/workshop-react-forms/67ebc1d57b0d863821af8ef2.md

```diff
@@ -0,0 +1,138 @@
+---
+id: 67ebc1d57b0d863821af8ef2
+title: Step 5
+challengeType: 0
+dashedName: step-5
+---
+
+# --description--
+
+Now, you will notice you can't type in the `input`. That's because you have to account for `setHeroName` setter variable too.
+
+To do that, use the `onChange` event to update `heroName` with the `input` value by using `setHeroName(e.target.value)`. This should be done in one line with an arrow syntax.
```

Comment:
We should update this to instead point to the warning logged in the console. That would be a nice lead into the `onChange`. Also the last sentence mentions the arrow function but doesn't mention that you need to pass in `e` for the parameter.

<img width="616" alt="Screenshot 2025-04-17 at 1 22 29 PM" src="https://github.com/user-attachments/assets/9d7772f1-438d-4fe7-ab8c-aa5d4ccddf35" />

```suggestion
Now, you will notice you can't type in the `input`. If you open up the console you will see a warning. 

To remove the warning, you will need to use the `onChange` event to update `heroName`. The value for the `onChange` event should be an arrow function with `e` for the parameter. The arrow function should implicitly return `setHeroName(e.target.value)`.
```